src/device-api/common.js

```diff
@@ -135,7 +135,7 @@ export function safeAppClone(app) {
 	const containerIdForService = _.fromPairs(
 		_.map(app.services, (svc) => [
 			svc.serviceName,
-			svc.containerId.substr(0, 12),
+			svc.containerId ? svc.containerId.substr(0, 12) : '',
```

Comment:
```suggestion
			svc.containerId != null ? svc.containerId.substr(0, 12) : '',
```
Just to make the intent clear

Reply:
What about
```suggestion
			svc.containerId?.substr(0, 12) ?? '',
```